lib/rules/no-eval.js

```diff
@@ -230,7 +195,12 @@ module.exports = {
                 "CallExpression:exit"(node) {
                     const callee = node.callee;
 
-                    if (isIdentifier(callee, "eval")) {
+                    /*
+                     * Optional call (`eval?.("code")`) is not direct eval.
+                     * The direct eval is only step 6.a.vi of https://tc39.es/ecma262/#sec-function-calls-runtime-semantics-evaluation
+                     * But the optional call is https://tc39.es/ecma262/#sec-optional-chaining-chain-evaluation
+                     */
```

Comment:
It works as direct eval in latest Chrome (could be Chrome bug?):

```js
var a = 5;
((a) => { eval?.("console.log(a)") })(10); // 10
```

Reply:
I reported:

- V8: https://bugs.chromium.org/p/v8/issues/detail?id=10630
- Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1647169
- test262: https://github.com/tc39/test262/pull/2667